main.cpp

```diff
@@ -179,6 +237,10 @@ void run()
   check_solution<T>(num_times, a, b, c, sum);
```

Comment:
I think this needs to be in an `#ifdef USE_MPI` section as well - otherwise you get massively spammed if verification fails.

On a related note, we really need to sort out verification for the reduction as per #20, since this almost always fails now when using a large number of MPI ranks.